### PR TITLE
Fix some npes in catalog service

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang PTAL, thanks

File: table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java [hunk 1]
```diff
@@ -248,7 +248,7 @@ HiveMetaStoreClient getHive() throws IOException {
       conf.set("hive.metastore.uris", mConfiguration.get(Property.HIVE_METASTORE_URIS));
       mHive = new HiveMetaStoreClient(conf);
       return mHive;
-    } catch (MetaException e) {
+    } catch (MetaException | NullPointerException e) {
```
Comment (yuzhu) [inline table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java]: hive client throws a NPE when the URI is not valid..

File: table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveUtils.java [hunk 2]
```diff
@@ -165,18 +170,22 @@ public static ColumnStatisticsInfo toProto(ColumnStatisticsObj colStats) {
...
                 .setMaxColLen(stringData.getMaxColLen())
-                .setNumNulls(stringData.getNumNulls()).setBitVectors(stringData.getBitVectors())
+                .setNumNulls(stringData.getNumNulls())
+                .setBitVectors(stringData.getBitVectors() == null
+                    ? "" : stringData.getBitVectors())
                 .build()).build());
       }
     }
-    if (statsData.isSetStringStats()) {
+    if (statsData.isSetBinaryStats()) {
```
Comment (gpang) [inline table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveUtils.java]: maybe a different PR, but can we have unittests for these types of bugs?

Comment (yuzhu) [reply table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveUtils.java]: will open a different PR

File: table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java (hunk 1 above)
Comment (yuzhu) [reply table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java]: added comment

**Comment by yuzhu** [comment]
@gpang PTAL, thanks

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by gpang** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**